clang/lib/CIR/CodeGen/CIRGenStmtOpenACC.cpp

```diff
@@ -96,18 +173,21 @@ mlir::LogicalResult CIRGenFunction::emitOpenACCOpAssociatedStmt(
 
 template <typename Op>
 mlir::LogicalResult
-CIRGenFunction::emitOpenACCOp(mlir::Location start,
+CIRGenFunction::emitOpenACCOp(OpenACCDirectiveKind dirKind,
+                              SourceLocation dirLoc, mlir::Location start,
```

Comment:
Why not use `mlir::Location` for `dirLoc`?

Reply:
Because that location is being used for NYI diagnostics, which take the Clang Source Location, not the mlir Location.

Reply:
I see. It's unfortunate to have a parameter that's just there for NYI diagnostics. I assume you plan to remove it when everything is implemented?

Reply:
Yep, exactly my intent.  I struggled a bit deciding how I could do the NYI without it, but figure I'll remove it once everythign is implemented.